c++/src/kj/compat/http.c++

```diff
@@ -2603,6 +2603,7 @@ public:
       if (fragments.empty()) {
         // This is the first fragment, so set the opcode.
         fragmentOpcode = opcode;
+        fragmentCompressed = recvHeader.isCompressed();
```

Comment:
Do you also need to reset `fragmentCompressed` here?:
```
        fragments.clear();
        fragmentOpcode = 0;
```

Reply:
Nice catch, yes it should reset. 

Practically speaking, I don't expect to see a situation where a connection uses compression for some messages and no compression for others. The chromium source code has a comment in it that suggests they could "predict" whether it would be worth it to compress some data, but IIRC it's been left as an unimplemented class. I haven't come across any libraries that let you intertwine (un)compressed messages either.

Very confused why the linux-musl test is failing, is there a way to get a stack trace?

Reply:
Might be the exception shown earlier in the output? :
```
kj/compat/http.c++:2447: failed: expected compressionConfig == nullptr [(can't stringify) == nullptr]; WebSocket compression is only supported if KJ is compiled with Zlib.
```

Don't know much about the musl build, but perhaps it lacks zlib?  If so, might need to wrap the new test with `#if KJ_HAS_ZLIB` .

Reply:
Ahhhh... totally slipped past me, thanks Joe, that must be it.

Reply:
Yeah, it doesn't help that the output appears interleaved with the earlier unrelated test results.  I assume that's an artifact of output buffering or github CI or something.

(FWIW, in reference to the earlier question, it's possible to dump stack trace addresses by adding a line like `KJ_DBG(kj::getStackTrace());` . But I don't know if that would necessarily have been useful here, since I'm not sure if there's an easy way set up to symbolize stack traces in github CI output.)